### DISCUSSION TITLE
Performance benchmarking for RTSP server as proxy

### DISCUSSION THREAD

**Discussion by vasanthi-crestron** [bluenviron/mediamtx#1261] [Questions]
Hi,  Looking for details on how many concurrent connections RTSP proxy server  can handle? I see "PROXY_COUNT" set to 20 in the config, file. 20 is the limit?  It seems very low.

**Comment by kdill00** [comment]
It will depend on what your doing. If your using the proxy, it’s just a pretty much straight copy with little overhead. I use it to proxy 25 security cameras, all being consumed in other applications or locations, some a few times over, and it only a consumes a core on an 80 core server that’s a few years old.   It uses ffmpeg stream copy and that’s it.   You will probably need to worry about network bandwidth before your computer runs out of processing power to proxy out more. It may eat up more ram im not sure.  I think even a raspberry pi 4 could proxy out 15 or more streams. If you need to do anything to the video it will get heavy fast without graphics cards or iGPU assistance.

**Comment by aler9** [comment]
> 25 security cameras, all being consumed in other applications or locations, some a few times over, and it only a consumes a core on an 80 core server that’s a few years old

:)

anyway, this server is used to proxy even 2000 cameras at once, it depends on the specific hardware.

[a comment by an automated account is not shown]
